addons/website/models/website_page.py

```diff
@@ -43,19 +43,20 @@ class Page(models.Model):
     website_id = fields.Many2one(related='view_id.website_id', store=True, readonly=False, ondelete='cascade')
     arch = fields.Text(related='view_id.arch', readonly=False, depends_context=('website_id',))
 
-    def _compute_homepage(self):
+    def _compute_is_homepage(self):
+        website = self.env['website'].get_current_website()
         for page in self:
-            page.is_homepage = page == self.env['website'].get_current_website().homepage_id
+            page.is_homepage = page.url == website.homepage_url
 
-    def _set_homepage(self):
+    def _set_is_homepage(self):
+        website = self.env['website'].get_current_website()
         for page in self:
-            website = self.env['website'].get_current_website()
             if page.is_homepage:
-                if website.homepage_id != page:
-                    website.write({'homepage_id': page.id})
+                if website.homepage_url != page.url:
+                    website.homepage_url = page.url
             else:
-                if website.homepage_id == page:
-                    website.write({'homepage_id': None})
+                if website.homepage_url == page.url:
+                    website.homepage_url = ''
```

Comment:
Yeah... I am too tired to know if this is right :') Seems too complicated, somehow a page whose is_homepage is written can update the website where the page does not even belong? I feel like writing is_homepage = True, should just be written as updating the homepage_url only once no?

Reply:
If called through something else than the page properties, yes.
But in that case, the `get_current_website` will already be incorrect, eg if done through a module install XML data.

Not ideal I guess but was always like that as we don't expect this to be set through unconventional flows and only through the page properties/manager.
Of course we could introduce a double check to ensure the page belongs to the website (meaning the page has no website or current_website_id).

But about the change itself, it actually doesn't change anything except it sets its url instead of its id.